File: Java Projects/Coffee Machine (Easy)/Stage 2/CoffeeMachine.java
package machine;

import java.util.Scanner;

public class CoffeeMachine {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Write how many cups of coffee you will need:");
        int numCups = scanner.nextInt();
        System.out.println("For " + numCups + " cups of coffee you will need:");
        System.out.println(numCups * 200 + " ml of water");
        System.out.println(numCups * 50 + " ml of milk");
        System.out.println(numCups * 15 + " g of coffee beans");
        
    }
}

File: Python Projects/Coffee Machine (Easy)/Stage 3/coffee_machine.py
ML_WATER_PER_CUP_COFFEE = 200
ML_MILK_PER_CUP_COFFEE = 50
G_BEANS_PER_CUP_COFFEE = 15

available_ml_water = 0
available_ml_milk = 0
available_g_beans = 0


def get_amount_supplies():
    global available_ml_water
    global available_ml_milk
    global available_g_beans

    print("Write how many ml of water the coffee machine has: ")
    available_ml_water = int(input())
    print("Write how many ml of milk the coffee machine has: ")
    available_ml_milk = int(input())
    print("Write how many grams of coffee beans the coffee machine has: ")
    available_g_beans = int(input())


def can_make_amount(num_cups):
    possible_cups = min([available_ml_water // ML_WATER_PER_CUP_COFFEE,
                         available_ml_milk // ML_MILK_PER_CUP_COFFEE,
                         available_g_beans // G_BEANS_PER_CUP_COFFEE])
    if possible_cups < num_cups:
        print(f"No, I can only make {possible_cups} of coffee")
    elif possible_cups == num_cups:
        print("Yes, I can make that amount of coffee")
    else:
        print(f"Yes, I can make that amount of coffee (and even {possible_cups - num_cups} more than that)")


get_amount_supplies()
print("Write how many cups of coffee you will need: ")
num = int(input())
can_make_amount(num)
